kazoo/tests/test_client.py

```diff
@@ -1160,6 +1165,118 @@ def test_request_queuing_session_expired(self):
         finally:
             client.stop()
 
+    def _require_zk_version(self, major, minor):
+        skip = False
+        if CI_ZK_VERSION and CI_ZK_VERSION < (major, minor):
+            skip = True
+        elif CI_ZK_VERSION and CI_ZK_VERSION >= (major, minor):
+            skip = False
+        else:
+            ver = self.client.server_version()
+            if ver[1] < minor:
+                skip = True
+        if skip:
+            pytest.skip("Must use Zookeeper %s.%s or above" % (major, minor))
+
+    def test_persistent_recursive_watch(self):
+        # This tests adding and removing a persistent recursive watch.
+        self._require_zk_version(3, 6)
+        events = []
+
+        def callback(event):
+            events.append(dict(type=event.type, path=event.path))
+
+        client = self.client
+        client.add_watch("/a", callback, AddWatchMode.PERSISTENT_RECURSIVE)
+        client.create("/a")
+        client.create("/a/b")
+        client.create("/a/b/c", value=b"1")
+        client.create("/a/b/d", value=b"1")
+        client.set("/a/b/c", value=b"2")
+        client.set("/a/b/d", value=b"2")
+        client.delete("/a", recursive=True)
+        # Remove the watch
+        client.remove_all_watches("/a", WatcherType.ANY)
+        # Perform one more call that we don't expect to see
+        client.create("/a")
+        # Wait in case the callback does arrive
+        time.sleep(3)
+        assert events == [
+            dict(type="CREATED", path="/a"),
```

Comment:
should we use `kazoo.protocol.states.EventType.CREATED` instead (and `CHANGED/DELETED` for the ones below?